agent_build/package_builders.py

```diff
@@ -930,20 +930,21 @@ def build(
         image_names = image_names or type(self).RESULT_IMAGE_NAMES
 
         for image_name in image_names:
-            for registry in registries:
-                for tag in tags:
-                    tag_options.append("-t")
-                    if registry:
-                        registry_prefix = f"{registry}/"
-                    else:
-                        registry_prefix = ""
-                    # NOTE: If we are using custom registry and image name already contains user
-                    # prefix, we remove it in case remove_image_prefix argument is provided (
-                    # this is desired in a lot of cases when using custom registry)
-                    # e.g. scalyr/scalyr-agent-2 -> scalyr-agent-2
-                    if remove_image_name_prefix and "/" in image_name:
-                        image_name = image_name.split("/")[1]
-                    tag_options.append(f"{registry_prefix}{image_name}:{tag}")
+
+            full_name = image_name
+
+            if user:
```

Comment:
I see you refactored that, good (previous approach with --remove-image-tag-name-prefix I did was a quick hack / workaround, this one should be better).